app/services/katello/pulp3/repository.rb

```diff
@@ -111,10 +111,23 @@ def get_distribution(href)
         nil
       end
 
+      def delete_distributions
+        paths.values.each do |path|
+          distribution_reference = distribution_reference(path)
+          if distribution_reference
+            delete_distribution(distribution_reference.href)
+            distribution_reference.destroy
+          end
+        end
+        []
+      end
+
       def delete_distributions_by_paths
         paths.values.each do |path|
           dists = lookup_distributions(base_path: path.sub(/^\//, ''))
-          delete_distribution(dists.first._href) if dists.first
+          if dists.first
+            delete_distribution(dists.first._href)
+          end
```

Comment:
Would it make sense to combine these methods?  delete_distributions_by_paths was used by the tests, but i think its okay to just modify it to do what delete_distributions is doing as well (return api output, delete the distribution reference)

Reply:
@jlsherrill yes, we certainly could. I wasn't sure if the delete...by_path had other intended uses.